### PR TITLE
Use docker-compose-tests.yml

### PR DESCRIPTION
**Pull request by abs51295**
- Use docker-compose-tests.yml instead of sed ... docker-compose.yml > docker-compose.dm-test.yml
- Resolves https://github.com/openshiftio/openshift.io/issues/3413

### CODE REVIEW

**Comment by msrb** [comment]
Hmm, do we need all those services here?

**Comment by abs51295** [comment]
@tuxdna might be the best person to answer that.

**Comment by abs51295** [comment]
I think I figured out which services are needed. Please review now @msrb ;)

File: docker-compose-tests.yml [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+     - postgres
+    ports:
+     - "5432:5432"
+    environment:
+      POSTGRESQL_USER: coreapi
+      POSTGRESQL_PASSWORD: coreapi
+      POSTGRESQL_DATABASE: coreapi
+      POSTGRESQL_INITIAL_DATABASE: postgres
+
+  worker-api: &worker
```
Comment (msrb) [inline docker-compose-tests.yml]: AFAIK, data-importer doesn't depend on workers, does it (?)

File: docker-compose-tests.yml [hunk 2]
```diff
@@ -0,0 +1,158 @@
+# NOTE: Containers use the generic "coreapi" prefix so they're independent of
+#       the particular project name. Project specific images are always built
+#       locally when using the docker-compose file.
+version: "2"
+services:
+  broker:
```
Comment (msrb) [inline docker-compose-tests.yml]: Do we need broker for data-model?

File: docker-compose-tests.yml [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  postgres:
+    image: registry.devshift.net/bayesian/coreapi-postgres
+    ports:
+     - "6432:5432"
+    environment:
+      POSTGRESQL_USER: coreapi
+      POSTGRESQL_PASSWORD: coreapi
+      POSTGRESQL_DATABASE: "coreapi"
+    container_name: coreapi-postgres
+  pgbouncer:
```
Comment (msrb) [inline docker-compose-tests.yml]: Do we need pgbouncer? :)

File: docker-compose-tests.yml [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+      BAYESIAN_DATA_IMPORTER_SERVICE_PORT: "9192"
+      SCANCODE_PROCESSES: "4"
+      # 'no-token' value forces the API call to not use ANY token.
+      # It works, but if abused, they can ban your IP.
+      # So if you need to analyse big number of packages locally,
+      # set this to your API Key (get it from https://libraries.io/account)
+      LIBRARIES_IO_TOKEN: "no-token"
+    tty: true  # yes, really -ti -d, binwalk chokes when there's no tty kept open
+
+  worker-ingestion:
```
Comment (msrb) [inline docker-compose-tests.yml]: Do we need this service?

File: docker-compose-tests.yml [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    <<: *worker
+    environment:
+      <<: *worker_environment
+      WORKER_ADMINISTRATION_REGION: ingestion
+
+  worker-db-migrations:
```
Comment (msrb) [inline docker-compose-tests.yml]: I guess the purpose of having this here is to create tables in posgtres (?)

**Comment by abs51295** [comment]
@msrb If you have a look at https://github.com/fabric8-analytics/fabric8-analytics-data-model/blob/master/runtests.sh#L22-L24, then it brings up `gremlin-http` and `worker-ingestion`. Now, if you look at what those deployments depend on in the `depends_on` section in `docker-compose-tests.yml` section, I think all the questions can be answered.

**Comment by msrb** [comment]
Oh my, so we clone deployment repository during tests?

**Comment by abs51295** [comment]
Unfortunately, yes.

**Comment by msrb** [comment]
> Unfortunately, yes. And that's what I removed in my changed runtests.sh, I took that file and copied into docker-compose-tests.yml by keeping the relevant parts.

Ok, this is bad. Everything is so tightly coupled together...

**Comment by abs51295** [comment]
By having the docker file inside the repository, it would not be coupled tightly as before, correct? We wouldn't be dependent on deployment repository any more to run tests.

**Comment by msrb** [comment]
This is going to be a lot of work: https://openshift.io/openshiftio/openshiftio/plan/detail/2336

**Pull request merged**